src/compiler/bundle/app-data-plugin.ts

```diff
@@ -182,7 +192,16 @@ const appendGlobalScripts = (globalScripts: GlobalScript[], s: MagicString) => {
   }
 };
 
-const appendBuildConditionals = (config: d.Config, build: d.BuildConditionals, s: MagicString) => {
+/**
+ * Generates the `BUILD` constant that is used at compile-time in a Stencil project
+ *
+ * **This function mutates the provided string argument**
```

Comment:
should this call out that it's a `MagicString` ? I didn't read the type signature and I was confused because I was like 'hey, strings are pass-by-value in JS, what is this witchcraft?'